src/snowflake/snowpark/mock/_connection.py

```diff
@@ -671,7 +668,7 @@ def execute(
 
         self.notify_query_listeners(QueryRecord("MOCK", "MOCK-PLAN"), **notify_kwargs)
 
-        return rows
+        return iter(rows) if to_iter else rows
```

Comment:
mock connection change -- previously it was returning before any of the AST data was sent.

Reply:
this change is done for both a.) mock and b.) regular connection>